src/extension.ts

```diff
@@ -67,16 +67,20 @@ async function pickDevice(): Promise<void> {
         // user canceled
         return;
     }
-    const statusBarMessage = new StatusBarProgressionMessage(`Connecting to device ${device.name}...`);
-    ev3devBrowserProvider.setDevice(device);
-    try {
-        await device.connect();
-        statusBarMessage.finish(`Successfully connected to device ${device.name}`);
-    }
-    catch (err) {
-        vscode.window.showErrorMessage(`Failed to connect to ${device.name}: ${err.message}`);
-        statusBarMessage.finish();
-    }
+    
+    vscode.window.withProgress({

```

Comment:
This has the same issues as https://github.com/ev3dev/vscode-ev3dev-browser/pull/22#discussion_r135401623

As it is, since you are not awaiting `withProgress`, you've changed the `pickDevice()` function so that the promise is resolved before we are actually connected.